docs/adrs/0077-permanently-store-origin-duty-gbloc.md

```diff
@@ -0,0 +1,107 @@
+---
+title: 0077 Permanently store Origin Duty Location GBLOC
+description: |
+  Decision outcome: Add a column to the orders table for Origin Duty Location GBLOC
+---
+
+# Permanently Store the Origin Duty Location GBLOC
+
+**User Story:** *[MB-15032](https://dp3.atlassian.net/browse/MB-15032)* :lock:
+
+## Problem Statement
+
+In our current database structure, GBLOCs are stored in the `postal_code_to_gblocs` as a one-to-many value: one GBLOC can represent many postal codes. Move origin duty locations associate to GBLOCs using the `origin_duty_location_to_gbloc` view. This view queries the `postal_code_to_gblocs` table on every read to fetch the GBLOC associated with the origin postal code on the orders while having the benefit of behaving like a table that can be joined in other queries.
+
+The downside of using a view to fetch the origin duty location GBLOC is that the data remains dynamic, and there is no way to determine what the *original* GBLOC was if the postal code to GBLOC relationship changes later. Since GBLOCs can be consolidated, USTC needs to be able to access the historical data for moves. 
```

Comment:
**question**: Since GBLOCs can be consolidated or changed, do we need a way in the system to track which GBLOCs are currently active? Maybe we already have something like that, tbh I haven't worked with GBLOCs much so there is plenty I don't know about this, but I don't see something in `postal_code_to_gblocs` or `origin_duty_location_to_gbloc`.

Reply:
Really good question, @felipe-lee . I'll see if I can get some information from Dan about this for the future. From what I can see, we don't have a tracking system in place for tracking active GBLOCs either, outside of searching migration scripts.